File: cmps012b/lab3/FileReverse.c
/* Tim Mertogul
 * 1331402
 * CMPS12B
 * 07/17/13
 * FileReverse Program that reverses the contents of an input file to an output file
 * FileReverse.java 
 */






#include<stdio.h>
#include<stdlib.h>
#include<string.h>

void stringReverse(char* s){

FILE* out; 
int i = 0;
int j = strlen(s)-1;
char temp[256];
   while(i<j){
      temp[0] = s[i];
      s[i] = s[j];
      s[j] = temp[0];
      i++;
      j--;
   }
   out = fopen("out", "w");
}


int main(int argc, char* argv[]){
   FILE* in; /* file handle for input */ 
   FILE* out; /* file handle for output */
   char word[256]; /* char array to store words from input file */

   /* check command line for correct number of arguments */
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   /* open input file for reading */
   in = fopen(argv[1], "r");
   if( in==NULL ){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   /* open output file for writing */
   out = fopen(argv[2], "w");
   if( out==NULL ){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   /* read words from input file, print on separate lines to output file*/
   while( fscanf(in, " %s", word) != EOF ){
      stringReverse(word);
      fprintf(out, "%s\n", word);
   }

   /* close input and output files */
   fclose(in);
   fclose(out);

   return(EXIT_SUCCESS);
}

File: cmps012b/lab2/FileReverse.java
/* Tim Mertogul
 * 1331402
 * CMPS12B
 * 07/09/13
 * FileReverse Program that reverses the contents of an input file to an output file
 * FileReverse.java 
 */

import java.io.*;
import java.util.Scanner;

class FileReverse{

   public static void main(String[] args) throws IOException{
      Scanner in = null;
      PrintWriter out = null;
      String line = null;
      String[] token = null;
      String str = null;
      int n, i, lineNumber = 0;
      if(args.length < 2){
         System.out.println("Usage: FileCopy infile outfile"); 
         System.exit(1);
      }
      in = new Scanner(new File(args[0]));
      out = new PrintWriter(new FileWriter(args[1]));
      in.useDelimiter("\n");
      
      while( in.hasNext() ){
         lineNumber++;
         line = in.next() + " "; // add space so split works on blank lines 
         token = line.split("\\s+"); // split line around white space
         n = token.length;
         for(i=0; i<n; i++){
            str = token[i];
            
            out.println(stringReverse(str, str.length()));
         }
      }
      in.close();
      out.close();
   }

   public static String stringReverse(String s, int n){
      if (n > 0) { return s.substring(n-1, n) + stringReverse(s, n-1); }
      else { return ""; }
   }
}
